packages/wrangler/src/config/validation.ts

```diff
@@ -670,20 +670,39 @@ function isValidRouteValue(item: unknown): boolean {
 	return false;
 }
 
+/**
+ * If account_id has been passed as an empty string, normalise it to undefined.
+ * This is to workaround older wrangler1-era templates that have account_id = '',
+ * which isn't a valid value anyway
+ */
+function mutateEmptyStringAccountIDValue(
+	diagnostics: Diagnostics,
+	rawEnv: RawEnvironment
+) {
+	if (rawEnv.account_id === "") {
+		diagnostics.warnings.push(
+			`The "account_id" field in your configuration is an empty string and will be ignored.\n` +
+				`Please remove the "account_id" field from your configuration.`
+		);
+		rawEnv.account_id = undefined;
+	}
+	return rawEnv;
+}
+
 /**
  * Normalize empty string to `undefined` by mutating rawEnv.route value.
  * As part of backward compatibility with Wrangler1 converting empty string to `undefined`
  */
 function mutateEmptyStringRouteValue(
-	rawEnv: RawEnvironment,
-	diagnostics: Diagnostics
+	diagnostics: Diagnostics,
+	rawEnv: RawEnvironment
 ): RawEnvironment {
 	if (rawEnv["route"] === "") {
 		diagnostics.warnings.push(
-			`Route assignment ${JSON.stringify(rawEnv["route"])} will be ignored.`
+			`The "route" field in your configuration is an empty string and will be ignored.\n` +
+				`Please remove the "route" field from your configuration.`
```

Comment:
Allowing the empty string, then asking them to remove the field seems ambivalent.

Reply:
It's more actionable than just saying it'll be ignored, this gives them an action to take to remove the warning. 
It'll happen when they use an older project or template. We'll have a go at cleaning up the templates when we work on `generate`, but we should also give actionable advice.